### PR TITLE
Add gradle scan plugin

### PR DESCRIPTION
**Pull request by danielgomezrico**
With it you can profile builds easy with a gradle param like:

```
# ./gradlew build -Dscan
```

and then you will see an Url with cool build info (better than `--profile`)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -100,4 +103,10 @@ subprojects {
             }
         }
     }
+}
+
+apply plugin: "com.gradle.build-scan"
```
Comment (artem-zinnatullin) [inline build.gradle]: nit: single quotes, they don't allow string interpolation

File: dependencies.gradle [hunk 2]
```diff
@@ -1,117 +1,113 @@
...
-        assertJ                 : "org.assertj:assertj-core:$versions.assertJ",
-        equalsVerifier          : "nl.jqno.equalsverifier:equalsverifier:$versions.equalsVerifier",
-        okHttpMockWebServer     : "com.squareup.okhttp3:mockwebserver:$versions.okHttp",
-        mockitoCore             : "org.mockito:mockito-core:$versions.mockito",
-        supportTestRunner       : "com.android.support.test:runner:$versions.supportTestRunner",
-        supportTestRules        : "com.android.support.test:rules:$versions.supportTestRunner",
-        espressoCore            : "com.android.support.test.espresso:espresso-core:$versions.espresso",
-        espressoContrib         : "com.android.support.test.espresso:espresso-contrib:$versions.espresso",
-]
+ext.versions = [code                         : 1,
```
Comment (artem-zinnatullin) [inline dependencies.gradle]: oh damn, GitHub is bad at such diffs…

**Comment by artem-zinnatullin** [comment]
Link to this feature won't hurt! https://scans.gradle.com/plugin

File: build.gradle [hunk 3]
```diff
@@ -41,6 +41,9 @@ buildscript {
 
         // Check latest version of dependencies, task: dependencyUpdates
         classpath gradlePlugins.versions
+
+        // Analyse builds to profile
+        classpath gradlePlugins.buildScan
```
Comment (artem-zinnatullin) [inline build.gradle]: Looks like this plugin eats too much memory and CI fails with error code `137` which means OutOfMemoryError.

Can you please try to tune memory usage to pass the CI?

File: dependencies.gradle (hunk 2 above)
Comment (vanniktech) [inline dependencies.gradle]: @artem-zinnatullin append `?=w1` to the link and it'll filter out diffs. [magic](https://github.com/artem-zinnatullin/qualitymatters/pull/202/files?w=1)

Comment (artem-zinnatullin) [inline dependencies.gradle]: 👍

**Comment by artem-zinnatullin** [comment]
Iteresting info https://gradle.com/s/u35qijqph5kku, thanks @danielgomezrico!

**Comment by danielgomezrico** [comment]
It says `32 cores` wow

**Comment by artem-zinnatullin** [comment]
Yup, but still fails because of OOM :(

**Comment by danielgomezrico** [comment]
@artem-zinnatullin can you share the build link?

**Comment by artem-zinnatullin** [comment]
Sure @danielgomezrico 

https://circleci.com/gh/artem-zinnatullin/qualitymatters/186, https://circleci.com/gh/artem-zinnatullin/qualitymatters/183

>Your build has exceeded the memory limit of 4G on 1 container. The results of this build are likely invalid. We have taken a snapshot of the memory usage at the time, which you can find in a build artifact named `memory-usage.txt`. The RSS column in this file shows the amount of memory used by each process, measured in kilobytes.

Looks like sometimes it fails so bad that it just prints "sh ci.sh died unexpectedly" or "Gradle build daemon disappeared unexpectedly (it may have been killed or may have crashed)".

But from time to time I see error code 137 which is OOM :(

**Comment by danielgomezrico** [comment]
@artem-zinnatullin Im seeing this one https://circleci.com/gh/artem-zinnatullin/qualitymatters/153 and looks like the same issue but before gradle scan code was in the repo, maybe is related to this?

**Comment by danielgomezrico** [comment]
lets continue in the open PR https://github.com/artem-zinnatullin/qualitymatters/pull/204 ?

**Pull request merged**